src_c/rect.c

```diff
@@ -588,15 +603,58 @@ pg_rect_collidepoint(pgRectObject *self, PyObject *args)
 }
 
 static PyObject *
-pg_rect_colliderect(pgRectObject *self, PyObject *args)
+pg_rect_colliderect(pgRectObject *self, PyObject *const *args,
+                    Py_ssize_t nargs)
```

Comment:
This PR adds a lot of complexity to this function, it should have a comment explaining the complexity.

Some potential content--
It's using fastcall so it doesn't have an intermediate tuple of arguments to deal with, so it needs to pull one level of pgRectFromObj logic up into itself dealing with the pyobjects directly.

^Though you probably have other things to add, and could write it out more formally than I did.